Example/Backpack/UIKit/Components/Carousel/CarouselViewController.swift

```diff
@@ -0,0 +1,92 @@
+//
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright © 2023 Skyscanner Ltd. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import UIKit
+import Backpack
+
+final class CarouselViewController: UIViewController {
+    
+    private let images = [
+        CarouselViewController.createImageView(with: "dialog_flare"),
+        CarouselViewController.createImageView(with: "dialog_image"),
+        CarouselViewController.createImageView(with: "pilanesburg-south-africa")
+    ]
+    
+    private lazy var carousel = BPKCarousel(
+        images: images,
+        currentImage: 1,
+        onImageChanged: onImageChanged
+    )
+    
+    private lazy var descriptionLabel: UILabel = {
+        let label = BPKLabel(fontStyle: .textCaption)
+        label.textColor = BPKColor.textPrimaryColor
+        label.text = "Currently selected index:"
+        return label
+    }()
+    
+    private lazy var displayedIndexLabel: UILabel = {
+        let label = BPKLabel(fontStyle: .textCaption)
+        label.textColor = BPKColor.textPrimaryColor
+        return label
+    }()
+    
+    private lazy var onImageChanged: ((Int) -> Void)? = { [weak self] selectedRowIndex in
+        self?.displayedIndexLabel.text = "\(selectedRowIndex)"
+    }
```

Comment:
this shouldn't be the way a closure is handled (refering to having it as `lazy var`). Lets see how we can change the closure/event/delegation system of the carousel.

I think delegate pattern works best for this kinds of components. We can have a chat about this!